### PR TITLE
Select: changing so we render the menu in a portal to fix issues with menu being hidden

### LINKED ISSUE #31826: Dropdown select field inside collapse, its not rendering properly, its options are hiding, its not extending beyond the collapsable area
Inside collapse component, when we click on the dropdown field, options are not visible, its rendering in the collapsable area, rest of the part is hidden.

Things are working fine in 7.3.7 Version, when we using 7.4.* its not working.

### LINKED ISSUE #32588: At row options available only first variable
**What happened**:

At row options available only first variable

image

image

**What you expected to happen**:

Must be available all variables

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**: 
- Grafana version: 7.5.1, 7.5.2
- Data source type & version: SQL Server
- OS Grafana is installed on: Win10
- User OS & Browser: Chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
In this PR I am moving the work from #32411 that has to do with changing the select menu to be rendered in a portal. I am moving this for better traceability and that it will close a number of issues.

More information: 
- https://github.com/grafana/grafana/pull/32411#issuecomment-810898906
- https://github.com/grafana/grafana/pull/32411#issuecomment-815520834

**Todo:**
- [ ] Hide the menu on scrolling.
- [ ] Make sure the menu is not "moving" on highlight.

**Which issue(s) this PR fixes**:
Fixes #32588
Fixes #31826

**Special notes for your reviewer**:
I just did a quick search but I think there are more issues that should be fixed by this.

### CODE REVIEW

**Comment by hugohaggmark** [comment]
@mckn what is the state of this PR, is it WIP? I'm asking because I can see there are todos that aren't checked yet. Can I help you close this?

**Comment by mckn** [comment]
> @mckn what is the state of this PR, is it WIP? I'm asking because I can see there are todos that aren't checked yet. Can I help you close this?

Yes please! It is WIP. I think both @dprokop and @jackw knows what to do be able to close this PR.

**Review by hugohaggmark — COMMENTED** [review]
Not sure I'll be able to fix this before v8

File: packages/grafana-ui/src/themes/default.ts [hunk 1]
```diff
@@ -25,6 +25,17 @@ export const commonColorsPalette = {
...
+  navbarFixed: 1000,
+  sidemenu: 1020,
+  dropdown: 1030,
+  typeahead: 1030,
+  tooltip: 1040,
+  modalBackdrop: 1050,
+  portal: 1051,
+  modal: 1060,
+};
+
```
[the remark this answers is not in the document]
Comment (torkelo) [reply packages/grafana-ui/src/themes/default.ts:38]: ah, ignore me.

**Comment by torkelo** [comment]
Updated the modal so that does not have overflow on it, https://github.com/grafana/grafana/pull/33939  this at least fixes the really critical problem with selects in modals

**Comment by ashharrison90** [comment]
superseded by https://github.com/grafana/grafana/pull/36398 🙌

**Pull request closed without merging**